witness/src/Dscp/Snowdrop/Actions.hs

```diff
@@ -40,23 +39,24 @@ initSDActions ::
        (MonadIO m, MonadCatch m, MonadIO n, MonadCatch n, HasWitnessConfig)
     => m (SDActionsM n)
 initSDActions = do
-    avlInitState <- initAVLPureStorage @Ids @Values initAccounts
-    (serverStDba, serverLookupHash) <- avlServerDbActions avlInitState
-    serverBlkDba <- blockDbActions
+    avlInitState <- liftIO $ initAVLPureStorage @Ids @Values initAccounts
+    (serverStDba, serverLookupHash) <- avlServerDbActions @Ids @Values @n avlInitState
+    serverBlkDba <- liftIO $ blockDbActions
     let sdActions = SDActionsM serverStDba serverBlkDba
 
     -- This is something to be used by AVL client (???)
-    let retrieveF :: AvlHash -> n (Maybe ByteString)
-        retrieveF h = serverLookupHash h >>= \resp -> do
+    let _retrieveF :: AvlHash -> m (Maybe ByteString)
+        _retrieveF h = serverLookupHash h >>= \resp -> do
           whenJust resp $ \resp' -> do
             (respV :: AVL.MapLayer AvlHash Ids Values AvlHash) <- deserialiseM resp'
             let respProof = AVL.Proof . Free $ pure <$> respV
             when (not $ AVL.checkProof h respProof) $ throwM BrokenProofError
           pure resp
-    let (rootHash :: RootHash) = gett avlInitState
+    let (_rootHash :: RootHash AvlHash) = gett avlInitState
 
     -- What am I supposed to do with it?
-    _clientStDba <- avlClientDbActions @Ids @Values retrieveF rootHash
+    -- pva701: it will be used in a AVL client, keep it compatible
```

Comment:
I know it's for "AVL client" so I keep it compatible, but I doubt we'll ever need AVL client.